### ISSUE TITLE
Smaller library size of vanilla bundle?

### ISSUE THREAD

**Issue by probil** [daybrush/moveable#15]
Any plans to make vanilla bundle smaller?
Currently it's around `70kb min`

I think there is a potential to make it `~10kb min`

**Comment by daybrush** [comment]
@probil 

I think my worries are similar.
Its size seems to grow as its functions become more and more.
However, there is an intention to reduce it without increasing it on schedule.

**Comment by deepakjoy** [comment]
Yes, it would be great to have some way to pick the components needed, or even some precompiled combinations as a start. Currently, the minified version is 145kb, which makes it too large to include for the limited functionality I need.

**Comment by jamesopti** [comment]
Surprised this doesnt have more votes/discussion

The core `react-moveable` bundle is quite large if youre only using one of the features, like resize.

https://bundlephobia.com/package/react-moveable@0.32.5

<img width="943" alt="Screen Shot 2022-04-30 at 8 57 44 AM" src="https://user-images.githubusercontent.com/3718812/166113029-7872aa2a-982f-44e3-b339-7c9edf926491.png">

**Comment by daybrush** [comment]
@jamesopti 
There is a way to support Tree Shacking only in react, and you can reduce the size.

```tsx
import * as React from "react";
import { makeMoveable, ResizableProps, Resizable } from "react-moveable";


// In order to use only some able, make a component with makeMoveable function.
const Moveable = makeMoveable<ResizableProps>([Resizable]);
```

**Comment by evanshe** [comment]
@daybrush is it available in svelte-moveable package?

**Comment by daybrush** [comment]
@evanshe

**Comment by daybrush** [comment]
@evanshe 
It will take time, but ultimately it will be a problem that will be solved.

**Comment by osbre** [comment]
@daybrush I think there is a way to make the Moveable library bundle much smaller by using `preact-compat` instead of React. Preact is a much lighter alternative to React which maintains compatibility with React API. Let me know what you think. Thank you 🙏

**Comment by daybrush** [comment]
@osbre 

I am using `croact`(like preact) which I developed.

I plan to reduce the size by tree shaking the unused able.
